docs/2_2_common_mechanisms.adoc

```diff
@@ -377,57 +382,51 @@ include::../headers/fmi3FunctionTypes.h[tags=SetClock]
 
-If <<get-and-set-variable-values,`fmi3Set{VariableType}`>> is called violating the min/max attribute settings of the corresponding variable, the following actions are performed:
+* During system creation and prototyping, checks on min/max should be performed.
+For maximum performance or real-time systems, these checks might not be performed.
 
-* On a <<fixed>> or <<tunable>> <<parameter>> <<fmi3Discard,`fmi3Status == fmi3Discard`>> is returned.
-* On an <<input>>, the FMU decides what to return (If no computation is possible, it could return <<fmi3Discard,`fmi3Status == fmi3Discard`>>, in other situations it may return <<fmi3Warning>> or <<fmi3Error>>, or <<fmi3OK>>, if it is uncritical).
+The approach in FMI is therefore that min/max definitions inform the importer about the region in which the FMU is designed to operate.
+However, the FMU should not rely on the min/max range to be properly observed.
+For example, dividing by an <<input>> or taking the square root of an <<input>> may result in returning either <<fmi3Discard>> or <<fmi3Error>>, or the FMU is able to handle this situation gracefully.
 
-If an FMU defines min/max values for integer types and `<Enumeration>` variables (<<local>> and <<output>> variables), then the expected behavior of the FMU is that <<get-and-set-variable-values,`fmi3Get{VariableType}`>> functions return values in the defined range.
+If an FMU defines min/max values for <<local>> and <<output>> variables of integer and <Enumeration> types, then the FMU must return values within the defined range when <<get-and-set-variable-values,`fmi3Get{VariableType}`>> is called.
 
-If an FMU defines min/max values, then the expected behavior of the FMU is that <<get-and-set-variable-values,`fmi3Get{VariableType}`>> returns values at the solution (accepted steps of the integrators) in the defined range with a certain uncertainty related to the tolerances of the numerical algorithms.]
+If an FMU defines min/max values for <<local>> and <<output>> variables of floating point types, then the FMU may return values in the defined range with a certain uncertainty related to the tolerances of the numerical algorithms and representational limits of IEEE 754.
 
 ===== Getting and Setting the Complete FMU State [[get-set-fmu-state]]
 
---
-The FMU has an internal state consisting of all values that are needed to continue a simulation.
+The FMU has internal data representing its state.
 This internal state consists especially of the values of the continuous states, iteration variables, <<parameter>> values, <<input>> values, delay buffers, file identifiers, and FMU internal status information.
-With the functions of this section, the internal FMU state can be copied and the pointer to this copy is returned to the environment.
-The FMU state copy can be set as actual FMU state, in order to continue the simulation from it.
+Depending on the FMI type, only a subset of this data is directly accessible via the FMI C-API.
```

Comment:
Is it FMI type or FMU type?

Reply:
or instance type?

Reply:
In the text we call it FMI type, or interface type, so I think functional mockup interface type makes most sense.